pandas/core/indexing.py

```diff
@@ -1073,8 +1085,9 @@ def _getitem_axis(self, key, axis=None):
         if axis is None:
             axis = self.axis or 0
 
-        if self._should_validate_iterable(axis):
-            self._has_valid_type(key, axis)
+        if is_iterator(key):
```

Comment:
is this is_iterator needed here?

Reply:
Didn't check this.

Reply:
It could be moved at a later stage... but I would see it as a loss (more general argument in #20748)